sale_pricelist_triple_discount/model/pricelist.py

```diff
@@ -1,22 +1,48 @@
 # Copyright 2019 Simone Rubino - Agile Business Group
+# Copyright (c) 2021 Andrea Cometa - Apulia Software s.r.l.
+# Copyright 2022 Alberto Re - Agile Business Group
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
-import openerp.addons.decimal_precision as dp
-from openerp import fields, models
+from odoo import fields, models
 
 
 class ProductPricelistItem(models.Model):
     _inherit = "product.pricelist.item"
 
     discount2 = fields.Float(
         "Discount 2 (%)",
-        digits_compute=dp.get_precision("Discount"),
+        digits="Discount",
         help="Second discount applied on a sale order line.",
         default=0.0,
     )
 
     discount3 = fields.Float(
         "Discount 3 (%)",
-        digits_compute=dp.get_precision("Discount"),
+        digits="Discount",
         help="Third discount applied on a sale order line.",
         default=0.0,
     )
+
+    def _compute_price(self, price, price_uom, product, quantity=1.0, partner=False):
+        """Compute the unit price of a product in the context of a pricelist
+        application. The unused parameters are there to make the full
+        context available for overrides.
+        """
+        self.ensure_one()
+
+        if self.compute_price != "formula":
+            return super()._compute_price(price, price_uom, product, quantity, partner)
+        else:
+            # Here we save price_round, then we temporarily unset it
```

Comment:
In the **formula** computation there are also margins and surcharge that are not considered here.
This leads to an unexpected behavior: if the user sets that he wants at least a 100 margin and applies the additional discounts, he would end up having less than a 100 margin.

As a end user I would expect that the discounts are applied at the same time as the formula's discount (`price_discount`).

Instead of reversing most of the formula, it is probably better to simplify the module behavior and only apply the additional discounts in the **percentage** case, what do you think?
See the proposed approach in https://github.com/OCA/sale-workflow/pull/2773.